shared/src/commonMain/kotlin/co/nimblehq/kmm/template/data/extensions/ResponseMapping.kt

```diff
@@ -0,0 +1,23 @@
+package co.nimblehq.kmm.template.data.extensions
+
+import co.nimblehq.jsonapi.model.JsonApiException
+import co.nimblehq.kmm.template.domain.exceptions.ApiException
+import kotlinx.coroutines.flow.*
+import kotlin.experimental.ExperimentalTypeInference
+
+@OptIn(ExperimentalTypeInference::class)
+fun <T> flowTransform(@BuilderInference block: suspend FlowCollector<T>.() -> T) = flow {
+    runCatching { block() }
+        .onSuccess { emit(it) }
+        .onFailure { throw it.mapError() }
+}
+
+private fun Throwable.mapError(): Throwable =
+    when (this) {
+        is JsonApiException -> ApiException(
+            code = errors.first().code,
+            message = errors.first().detail,
+            cause = this
+        )
+        else -> this
+    }
```

Comment:
Please ignore implementing JSON:API from this PR 🙏 

```suggestion
private fun Throwable.mapError(): Throwable =
    when (this) {
    	// TODO catch custom errors
        else -> this
    }
```